### PR TITLE
Database

### PR DESCRIPTION
**Pull request by shell02**
Many general queries created in models (might need to add more as we go)
Database Service created to find possible errors before and after making the queries

We should not interact with the models directly but go through the Database Service

### CODE REVIEW

**Comment by shell02** [comment]
I think the docker-compose does not work with express uncommented, will do another commit later

For now, just do docker-compose with postgres and pgweb and go inside API directory and "npm run dev"

File: API/src/models/Notification.ts [hunk 1]
```diff
@@ -1,49 +1,120 @@
...
+ * username: string
+ * type: NotificationType
+ * notifid: number
+ */
+interface NotificationQueryResult {
+  touserid: number
+  fromuserid: number
+  username: string
+  type: NotificationType
+  notifid: number
```
Comment (tidurand) [inline API/src/models/Notification.ts:34]: Why not use camelCase ?
For others interfaces too

File: API/src/models/Tag.ts [hunk 2]
```diff
@@ -1,56 +1,136 @@
...
       .query(`DELETE FROM tag WHERE content = $1`, [content])
-      .then((res) => (res.rows.length > 0 ? res.rows[0] : null));
-  },
-};
+      .then((res) => (res.rowCount > 0 ? true : false))
+      .catch(() => false)
+  }
+}
+
+// db.exists(tableName)
```
Comment (tidurand) [inline API/src/models/Tag.ts]: For all comments code

It is because we will use later ?

File: API/src/models/Notification.ts (hunk 1 above)
Comment (shell02) [reply API/src/models/Notification.ts:34]: the response to the query is not in camelCase (postgresql is case insensitive). 

I transform the response to camelCase with the function toNotificationRow(row: NotificationQueryResult): NotificationRow {}

**Review by tidurand — APPROVED** [review]

File: API/src/models/Tag.ts (hunk 2 above)
Comment (shell02) [reply API/src/models/Tag.ts]: I deleted all of these comments i forgot about them in my first commit. They are useless now

**Pull request merged**